test/suites/standard2/workload-partitioning.robot

```diff
@@ -1,21 +1,165 @@
+    ...    \ \ \ \ full-pcpus-only: "true"
+    ...    \ \ cpuManagerReconcilePeriod: 5s
+
+    ${merged}=    Extend MicroShift Config    ${kubelet_config}
+    Upload MicroShift Config    ${merged}
+
+Configure CRIO For Workload Partitioning
+    [Documentation]    add crio Dropin configuration
+    [Arguments]    ${cpus}
+    ${crio_configuration}=    CATENATE    SEPARATOR=\n
+    ...    [crio.runtime]
+    ...    infra_ctr_cpuset = "${cpus}"
+    ...    [crio.runtime.workloads.management]
+    ...    activation_annotation = "target.workload.openshift.io/management"
+    ...    annotation_prefix = "resources.workload.openshift.io"
+    ...    resources = { "cpushares" = 0, "cpuset" = "${cpus}" }
+
+    Upload String To File    ${crio_configuration}    ${CRIO_CONFIG_DROPIN}
+
+Configure CPUAffinity In Systemd
+    [Documentation]    add CPUAffinity in systemd unit dropin
+    [Arguments]    ${cpus}    ${target_file}
+    Create Remote Dir For Path    ${target_file}
+    ${systemd_cpu_affinity}=    CATENATE    SEPARATOR=\n
+    ...    [Service]
+    ...    CPUAffinity=${cpus}
+    Upload String To File    ${systemd_cpu_affinity}    ${target_file}
+
+All Management Pods Should Run On Reserved CPU
+    [Documentation]    Verify all the PODs runs on explicit reserved CPU
+    [Arguments]    ${cpus}    ${is_workloads}=${EMPTY}
+
+    Set Global Variable    ${NOT_WORKLOADS}    ${EMPTY}
+    IF    "${is_workloads}"!="${EMPTY}"
+        Set Global Variable    ${NOT_WORKLOADS}    | not
+    END
+
+    ${stdout}    ${stderr}    ${rc}=    Execute Command
+    ...    crictl ps -q | xargs sudo crictl inspect | jq -rs '[.[] | select(.info.runtimeSpec.annotations["target.workload.openshift.io/management"] ${NOT_WORKLOADS})]'
+    ...    sudo=True
+    ...    return_stdout=True
+    ...    return_stderr=True
+    ...    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+
+    ${json_status}=    Json Parse    ${stdout}
+    FOR    ${pod}    IN    @{json_status}
+        ${pod_info}=    Catenate    SEPARATOR=
+        ...    container: ${pod}[info][runtimeSpec][annotations][io.kubernetes.pod.name]
+        ...    ${EMPTY} pod: ${pod}[status][metadata][name]
+        ...    ${EMPTY} pid: ${pod}[info][pid]
+        ...    ${EMPTY} namespace: ${pod}[info][runtimeSpec][annotations][io.kubernetes.pod.namespace]
+        IF    "${is_workloads}"=="${EMPTY}"
+            IF    ${pod}[info][runtimeSpec][linux][resources][cpu][cpus] != ${cpus}
+                Fail
+                ...    Management ${pod_info} running on CPU ${pod}[info][runtimeSpec][linux][resources][cpu][cpus] instead of ${cpus}
+            END
+            Proccess Should Be Running On Host CPU    ${pod}[info][pid]    ${cpus}
+        ELSE IF    "${pod}[info][runtimeSpec][linux][resources][cpu][cpus]" == "${cpus}"
+            Fail
+            ...    Workload ${pod_info} running on CPU ${pod}[info][runtimeSpec][linux][resources][cpu][cpus] instead of ${cpus}
+        END
+    END
```

Comment:
We can use python function in RF. Do you think it could benefit in terms of readability?